cmd/template-resolver/utils/templateResolver.go

```diff
@@ -0,0 +1,337 @@
+package templateresolver
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"gopkg.in/yaml.v3"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/stolostron/go-template-utils/v5/pkg/templates"
+)
+
+func ProcessTemplate(yamlFile, hubKubeConfigPath, clusterName string) {
```

Comment:
Perhaps the YAML bytes can be passed to `ProcessTemplate` and the `main` function can be responsible for parsing the user input from the file path or stdin. This makes this API more flexible.

Reply:
Good idea--I wound up adding a `HandleFile()` function instead since I'm thinking it'd be useful for `acm-cli` to consume.